### PR TITLE
fix double count bug transfers for `zksync`, update native contract address on other chains

### PR DESCRIPTION
**Pull request by lgingerich**
### Update!
Please build spells in the proper [subproject](../dbt_subprojects/) directory. For more information, please see the main [readme](../README.md), which also links to a GH discussion with the option to ask questions.

### Contribution type
Please check the type of contribution this pull request is for:

- [ ] New spell(s)
- [X] Adding to existing spell lineage
- [ ] Bug fix

---

### For adding to existing spell lineage
If you are adding to an existing spell lineage, please provide the following information:

- **Description:**
This PR is intended to add token balances for ZKsync Era. In order to do so, there is first one necessary bug to fix in `transfers_base`. Below I describe this fix, another minor change for Linea, Scroll, and Polygon zkEVM, as well as the work to add balances for zksync

- **`transfers_base` macro:** Since ZKsync Era emits a transfer event for ETH (it essentially acts as an ERC20 token), some transfers get double counted during the process of unioning transfers found from traces and from erc20 transfer events.
  - To handle this scenario, I've added a `include_traces` param to the `transfers_base` macro. AFAIK, zksync is the only chain that currently emits events for ETH transfers, but this param may become more relevant if [EIP-7708](https://ethereum-magicians.org/t/eip-7708-eth-transfers-emit-a-log/20034) is approved.
    - If `include_traces = true`, the macro functionality remains the same as now. `include_traces = false`, only transfers from `erc20{chain}_evt_transfer get` included.

- **`native_contract_address` param:** Linea, Scroll, and Polygon zkEVM each had `native_contract_address = 0x000000000000000000000000000000000000800a`. This looks like a remnant of someone copying the `tokens_zksync_base_transfers.sql` implementation. I couldn't find any evidence of this native ETH token address existing for those chains, so I've set their default to `null`.

- **Adding balances:** In Work

### CODE REVIEW

**Comment by lgingerich** [comment]
Example Dune query to see the overcounting of ETH transfers on zksync:

```
SELECT
    'erc20_zkevm.evt_transfer' AS "table"
    , COUNT(*) AS eth_transfer_count
FROM erc20_zksync.evt_transfer
WHERE contract_address = 0x000000000000000000000000000000000000800a
AND DATE(evt_block_time) = DATE '2024-01-01'

UNION ALL

SELECT
    'tokens_zksync.transfers' AS "table"
    , COUNT(*) AS eth_transfer_count
FROM tokens_zksync.transfers
WHERE contract_address = 0x000000000000000000000000000000000000800a
AND DATE(block_time) = DATE '2024-01-01'
```

Output:
```

table | eth_transfer_count
-- | --
tokens_zksync.transfers | 6827239
erc20_zkevm.evt_transfer | 3984173
```

**Comment by lgingerich** [comment]
@aalan3 I was trying to add balances for zksync but then found the discussion in https://github.com/duneanalytics/spellbook/issues/5123 about balances being built from raw data gathered outside of Spellbook. What is the process to get zksync added to this?

File: sources/_sector/tokens/balances_sources.yml [hunk 1]
```diff
@@ -135,3 +135,18 @@ sources:
           - name: token_id
           - name: amount
           - name: type
+
+  - name: tokens_zksync
+    description: "Prices tables across blockchains"
+    tables:
+      - name: balances_zksync
```
Comment (Hosuke) [inline sources/_sector/tokens/balances_sources.yml:142]: Hi @jeff-dude, do you know how to add `tokens_*.balances_*` for new chains?

**Comment by jeff-dude** [comment]
> @aalan3 I was trying to add balances for zksync but then found the discussion in #5123 about balances being built from raw data gathered outside of Spellbook. What is the process to get zksync added to this?

we had been prioritizing per chain, but have also paused new chains atm. i will have to raise internally and see. for now, can you remove balances spells from this PR and focus on transfers only?

File: sources/_sector/tokens/balances_sources.yml (hunk 1 above)
Comment (jeff-dude) [reply sources/_sector/tokens/balances_sources.yml:142]: @lgingerich can you remove source file changes too?

File: dbt_subprojects/tokens/models/transfers_and_balances/zksync/tokens_zksync_base_transfers.sql [hunk 2]
```diff
@@ -15,6 +15,7 @@
     traces = source('zksync','traces'),
     transactions = source('zksync','transactions'),
     erc20_transfers = source('erc20_zksync','evt_transfer'),
-    native_contract_address = '0x000000000000000000000000000000000000800a'
+    native_contract_address = '0x000000000000000000000000000000000000800a',
+    include_traces = false
```
Comment (jeff-dude) [inline dbt_subprojects/tokens/models/transfers_and_balances/zksync/tokens_zksync_base_transfers.sql:19]: had to remove single quotes on this value to make it work (tested locally by running `dbt compile` to see raw sql)

**Review by jeff-dude — APPROVED** [review]
went ahead and removed param from all other chains since it defaults in the macro, then the below comment. otherwise, i think i'm good here

**Pull request merged**